Entities/FloatierSpaceBlock.cs

```diff
@@ -0,0 +1,98 @@
+        }
+
+        public static void Unload() {
+            On.Monocle.Tracker.Initialize -= InitializeTracker;
+        }
+
+        private static void InitializeTracker(On.Monocle.Tracker.orig_Initialize orig) {
+            orig();
+            Tracker.TrackedEntityTypes[typeof(FloatierSpaceBlock)].Add(typeof(FloatySpaceBlock));
+        }
+
+        public override void Update() {
+            bool masterOfGroup = MasterOfGroup;
+            // If MasterOfGroup is false, FloatySpaceBlock does almost nothing in Update()
+            MasterOfGroupInfo.SetValue(this, false);
+            base.Update();
+            MasterOfGroupInfo.SetValue(this, masterOfGroup);
+
+            if (MasterOfGroup) {
+                bool playerRiding = false;
+                foreach (FloatySpaceBlock item in Group) {
+                    if (item.HasPlayerRider()) {
+                        playerRiding = true;
+                        break;
+                    }
+                }
+                if (!playerRiding) {
+                    foreach (JumpThru jumpthru in Jumpthrus) {
+                        if (jumpthru.HasPlayerRider()) {
+                            playerRiding = true;
+                            break;
+                        }
+                    }
+                }
+                float sinkTimerVal = (float) sinkTimerInfo.GetValue(this);
+                if (playerRiding) {
+                    sinkTimerInfo.SetValue(this, sinkTimerVal = 0.3f * floatinessBoost);
+                } else if (sinkTimerVal > 0f) {
+                    sinkTimerInfo.SetValue(this, sinkTimerVal -= Engine.DeltaTime);
+                }
+                float yLerpVal = (float)yLerpInfo.GetValue(this);
+                if (sinkTimerVal > 0f) {
+                    yLerpInfo.SetValue(this, Calc.Approach(yLerpVal, 1f, (1f / floatinessBoost) * Engine.DeltaTime));
+                } else {
+                    yLerpInfo.SetValue(this, Calc.Approach(yLerpVal, 0f, (1f / floatinessBoost) * Engine.DeltaTime));
+                }
+                sineWaveInfo.SetValue(this, (float)sineWaveInfo.GetValue(this) + Engine.DeltaTime);
+                dashEaseInfo.SetValue(this, Calc.Approach((float)dashEaseInfo.GetValue(this), 0f, Engine.DeltaTime * 1.5f));
+                MoveToTarget();
+            }
+            LiftSpeed = Vector2.Zero;
+        }
+
+        private void MoveToTarget() {
+            float sineWavePos = (float) Math.Sin((float)sineWaveInfo.GetValue(this)) * 4f;
+            Vector2 dashOffset = Calc.YoYo(Ease.QuadIn((float)dashEaseInfo.GetValue(this))) * (Vector2)dashDirectionInfo.GetValue(this) * 8f;
+            for (int i = 0; i < 2; i++) {
+                foreach (KeyValuePair<Platform, Vector2> move in Moves) {
+                    Platform key = move.Key;
+                    bool flag = (key is JumpThru jumpThru && jumpThru.HasRider()) || (key is Solid solid && solid.HasRider());
```

Comment:
Those two variable names could be improved on, but they're not a crime, so they're fine.